test/api/unit/authorizers/site.test.js

```diff
@@ -1,216 +1,272 @@
+      const [user, org, role] = await Promise.all([
+        factory.user(),
+        factory.organization.create(),
+        factory.role.create(),
+      ]);
       const params = {
-        owner: crypto.randomBytes(3).toString("hex"),
-        repository: crypto.randomBytes(3).toString("hex"),
-        defaultBranch: "main",
-        engine: "jekyll",
+        owner: crypto.randomBytes(3).toString('hex'),
+        repository: crypto.randomBytes(3).toString('hex'),
+        defaultBranch: 'main',
+        engine: 'jekyll',
+        organizationId: org.id,
+      };
+
+      await org.addUser(user, { through: { roleId: role.id } });
+      const expected = await authorizer.create(user, params);
+
+      return expect(expected).to.be.undefined;
+    });
+
+    // eslint-disable-next-line consistent-return
+    it('should throw an error for user with organizations and no organizationId specified', async () => {
+      const [user, org, role] = await Promise.all([
+        factory.user(),
+        factory.organization.create(),
+        factory.role.create(),
+      ]);
+      const params = {
+        owner: crypto.randomBytes(3).toString('hex'),
+        repository: crypto.randomBytes(3).toString('hex'),
+        defaultBranch: 'main',
+        engine: 'jekyll',
+      };
+
+      await org.addUser(user, { through: { roleId: role.id } });
+
+      try {
+        await authorizer.create(user, params);
+      } catch (error) {
+        return expect(error.message).to.equal(siteErrors.ORGANIZATION_REQUIRED);
+      }
+    });
+
+    // eslint-disable-next-line consistent-return
+    it('should throw an error for user trying to add a site to an org they do not belong to', async () => {
+      const [user, org, role] = await Promise.all([
+        factory.user(),
+        factory.organization.create(),
+        factory.role.create(),
+      ]);
+      const params = {
+        owner: crypto.randomBytes(3).toString('hex'),
+        repository: crypto.randomBytes(3).toString('hex'),
+        defaultBranch: 'main',
+        engine: 'jekyll',
+        organizationId: 'not-their-org',
+      };
+
+      await org.addUser(user, { through: { roleId: role.id } });
+
+      try {
+        await authorizer.create(user, params);
+      } catch (error) {
+        expect(error.status).to.equal(404);
+        return expect(error.message).to.equal(siteErrors.NO_ASSOCIATED_ORGANIZATION);
+      }
+    });
+  });
+
+  describe('.findOne(user, site)', () => {
+    it('should resolve if the user is associated with the site', async () => {
+      const user = await factory.user();
+      const site = await factory.site({ users: Promise.all([user]) });
+      const expected = await authorizer.findOne(user, site);
+
+      return expect(expected).to.equal(site.id);
+    });
+
+    // eslint-disable-next-line consistent-return
+    it('should reject if the user is not associated with the site', async () => {
+      const user = await factory.user();
```

Comment:
could `Promise.all` here to speed it up! Obvi one of thees doesn't make a different, but in the aggregate it can.